testsuite/gnat2goto/tests/volatile/volatile_type.adb

```diff
@@ -0,0 +1,16 @@
+procedure Volatile_Type is
+   Data: Integer;
+   pragma Volatile(Data);
+   Flag: Boolean;
+   pragma Volatile(Flag);
+begin
+   Data := 42;
+   Flag := True;
+
+   if not Flag then
+      Data := 0;
+   end if;
+
+   pragma Assert (Data=42);
```

Comment:
What was the behaviour of symex prior to the previous changes? Would this particular test have failed anyway?

Reply:
No, it would look exactly the same way. Only the symbol table would contain the `volatile` flag and `gnat2goto` would complain about not supporting pragma volatile.